tests/System.Text.Primitives.Tests/Encoding/Utf8Utf16ConversionTests.cs

```diff
@@ -78,6 +78,127 @@ public void EnumerateAndEnsureCodePointsOfTheSameUtf8AndUtf16StringsAreTheSame(s
             }
         }
 
+        public static object[][] InvalidUtf8ToUtf16SequenceData = {
```

Comment:
I don't think these specific tests can be generalized more because they know about the underlying implementation. If we were to find an implementation with similar performance characteristics that didn't have to have various special paths through the code, then we wouldn't need some of these specialized tests.